clients/apple/Shared/Stateful Logic/FileService.swift

```diff
@@ -3,12 +3,68 @@ import SwiftLockbookCore
 
 class FileService: ObservableObject {
     let core: LockbookApi
+    
+    var successfulAction: FileAction? = nil
 
     @Published var root: File? = nil
     @Published var files: [File] = []
-    var successfulAction: FileAction? = nil
+    
+    // File Service keeps track of the children and parent being displayed on iOS. Since this functionality is not used for macOS, it is conditionally compiled.
+#if os(iOS)
+    @Published var parent: File? = nil
+    @Published var children: [File] = []
```

Comment:
can we keep track of less state in general? Couldn't we use `childrenOf` from below as we did before?